### PR TITLE
Np 3543 handler and client person inst info

### PR DESCRIPTION
**Pull request by torarnet**

### CODE REVIEW

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        apiClient = new FetchPersonInstitutionInfoClient(clientMock);
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new FetchPersonInstitutionInfoHandler(apiClient, environment);
+    }
+
+    @Test
+    void shouldReturnCorrectDataWhenGeneratingResponse() throws IOException, InterruptedException {
+        CristinPersonInstitutionInfo cristinInfo = new CristinPersonInstitutionInfo();
+        cristinInfo.setEmail(DUMMY_EMAIL);
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:76]: cristinInfo.setEmail(randomString())

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        cristinInfo.setPhone(DUMMY_PHONE);
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:77]: ditto

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    void setUp() throws IOException, InterruptedException {
+        when(clientMock.<String>send(any(), any())).thenReturn(new HttpResponseFaker(EMPTY_JSON, 200));
+        apiClient = new FetchPersonInstitutionInfoClient(clientMock);
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new FetchPersonInstitutionInfoHandler(apiClient, environment);
+    }
+
+    @Test
+    void shouldReturnCorrectDataWhenGeneratingResponse() throws IOException, InterruptedException {
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:74]: what is the correct data?

**Review by svenngjeto — DISMISSED** [review]

**Review by brinxmat — DISMISSED** [review]

**Review by svenngjeto — DISMISSED** [review]

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            sendQuery(null, Map.of(PERSON_ID, VALID_PERSON_ID, ORG_ID, VALID_INSTITUTION_ID));
+
+        PersonInstitutionInfo actual = gatewayResponse.getBodyObject(PersonInstitutionInfo.class);
+        PersonInstitutionInfo expected = cristinInfo.toPersonInstitutionInfo(EXPECTED_ID_URI);
+
+        assertThat(actual, equalTo(expected));
+    }
+
+    @Test
+    void shouldGenerateCorrectCristinUriFromPathParams() throws ApiGatewayException, IOException {
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:95]: define correct

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        assertEquals(HttpURLConnection.HTTP_BAD_REQUEST, gatewayResponse.getStatusCode());
+        assertEquals(APPLICATION_PROBLEM_JSON.toString(), gatewayResponse.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        assertThat(gatewayResponse.getBody(), containsString(INVALID_ORGANIZATION_ID));
+    }
+
+    private URI getCristinUriFromIdentifiers() {
+        return URI.create(EXPECTED_CRISTIN_URI);
+    }
+
+    private GatewayResponse<PersonInstitutionInfo> sendQuery(Map<String, String> queryParams,
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:155]: the query parameters are always null

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        handler.handleRequest(input, output, context);
+        return GatewayResponse.fromOutputStream(output);
+    }
+
+    private InputStream requestWithParams(Map<String, String> queryParams, Map<String, String> pathParams)
+        throws JsonProcessingException {
+
+        return new HandlerRequestBuilder<Void>(OBJECT_MAPPER)
+            .withBody(null)
+            .withQueryParameters(queryParams)
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:169]: queryParameters are never sent. Remove if not in use

Comment (axthosarouris) [reply cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:169]: If the intention is to validated that no query parameters are being sent, this should be an explicit test.

File: cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    void setUp() throws IOException, InterruptedException {
+        when(clientMock.<String>send(any(), any())).thenReturn(new HttpResponseFaker(EMPTY_JSON, 200));
+        apiClient = new FetchPersonInstitutionInfoClient(clientMock);
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new FetchPersonInstitutionInfoHandler(apiClient, environment);
+    }
+
+    @Test
+    void shouldReturnCristinResponseMappedToNvaFormatWhenCallingEndpointWithValidParams()
```
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandlerTest.java:73]: I think the description of this test is very generic. if you are testing the absence of query parameters then test should be called that. If you are testing the response then define what is the valid query.

File: cristin-person/src/main/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandler.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+    private final transient FetchPersonInstitutionInfoClient apiClient;
+
+    @JacocoGenerated
+    public FetchPersonInstitutionInfoHandler() {
+        this(new Environment());
+    }
+
+    @JacocoGenerated
+    public FetchPersonInstitutionInfoHandler(Environment environment) {
```
Comment (axthosarouris) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandler.java:35]: This constructor is used only by the constructor above

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: cristin-person/src/main/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandler.java (hunk 8 above)
Comment (torarnet) [reply cristin-person/src/main/java/no/unit/nva/cristin/person/institution/fetch/FetchPersonInstitutionInfoHandler.java:35]: I think we needed constructors like these for the template to call the handler?

**Review by axthosarouris — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**